relay-statsd/src/lib.rs

```diff
@@ -85,6 +96,23 @@ pub struct MetricsClient {
     pub rx: Option<crossbeam_channel::Receiver<Vec<u8>>>,
 }
 
+/// Client configuration used for initialization of [`MetricsClient`].
+#[derive(Debug)]
+pub struct MetricsClientConfig<'a, A: ToSocketAddrs> {
```

Comment:
Usually we do not add bounds to the struct, only on where it is used. Much easier to handle than doing it this way, the only exceptions really is if the `Drop` implementation requires it.